isolate_example/lib/page_three.dart

```diff
@@ -11,14 +11,254 @@
+    if (running) return;
+
+    running = true;
+    _timer.reset();
+    currentProgress.clear();
+
+    _timer.start();
+    _newIceSP.send('start');
+
+    notifyListeners();
+  }
+
+  Future<void> generateRandomNumbers() async {
+    if (running) return;
+
+    running = true;
+    Random rng = Random();
+    _timer = Stopwatch();
+
+    currentProgress.clear();
+    _timer.start();
+
+    List<int> randNums = [];
+    for (int i = 0; i < 100; i++) {
+      randNums.clear();
+
+      for (int j = 0; j < 1000000; j++) {
+        randNums.add(rng.nextInt(100));
+      }
+      await sendNumbers(randNums);
+    }
+  }
+
+  Future<void> sendNumbers(List<int> numList) async {
+    return Future<void>(() {
+      _newIceSP.send(numList);
+    });
+  }
+
+  Isolate get newIsolate => _newIsolate;
+
+  void dispose() {
+    super.dispose();
+    _newIsolate?.kill(priority: Isolate.immediate);
+    _newIsolate = null;
+  }
+}
+
+class RunningList extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    return Center(
-      child: Text('Welcome to the Data Transfer Page'),
+    final controller = Provider.of<DataTransferIsolateController>(context);
+    final progress = controller.currentProgress;
+
+    return DecoratedBox(
+      decoration: BoxDecoration(
+        color: Colors.grey[200],
+      ),
+      child: ListView.builder(
+        itemCount: progress.length,
+        itemBuilder: (context, index) {
+          return Column(
+            children: [
+              Card(
+                child: ListTile(
+                  title: Text(progress[index]),
+                ),
+                color: Colors.lightGreenAccent,
+              ),
+              Divider(
+                color: Colors.blue,
+                height: 3,
+              ),
+            ],
+          );
+        },
+      ),
     );
   }
 }
+
+Future<void> _secondIsolateEntryPoint(SendPort callerSP) async {
+  final randNums = <int>[];
+  ReceivePort newIceRP = ReceivePort();
+  callerSP.send(newIceRP.sendPort);
+
+  newIceRP.listen(
+    (dynamic message) async {
+      if (message is String && message == 'start') {
+        await generateAndSum(callerSP, createNums());
+
+        callerSP.send('done');
+      } else {
+        randNums.addAll(message as List<int>);
+
+        callerSP.send(randNums.length ~/ 1000000);
+
+        if (randNums.length == 100000000) {
+          addUp(randNums);
+          callerSP.send('done');
+          randNums.clear();
+        }
+      }
+    },
+  );
+}
+
+Iterable<int> createNums() sync* {
+  final rng = Random();
+  for (int i = 0; i < 100000000; i++) {
+    yield rng.nextInt(100);
+  }
+}
+
+Future<void> generateAndSum(SendPort callerSP, Iterable<int> iter) async {
+  int sum = 0;
+  int count = 0;
+
+  for (int x in iter) {
+    sum += x;
+    count++;
+    if ((count + 1) % 1000000 == 0) {
+      callerSP.send((count + 1) ~/ 1000000);
+    }
+  }
+
+  return sum;
+}
+
+int addUp(List<int> numList) {
```

Comment:
I think something has gotten mixed up since we talked on Wednesday. The idea of using createNums to create a list on the fly is that `generateAndSum` could be used for both message types (whether the main isolate provided a list of numbers or the second isolate generated them itself). That way there's no need for a second "add up the numbers and report progress" method like `addUp`.